src/components/search/SearchBar.vue

```diff
@@ -0,0 +1,82 @@
+<template>
+  <div class="search-box-wrapper">
+    <input
+      id="search-box"
+      v-model="searchQuery"
+      :placeholder="placeholder"
+      @keyup.enter="search(searchQuery)"
+    />
+    <div :class="searchQuery ? 'hidden' : 'visible'">
+      <img
+        svg-inline
+        class="icon icon-search"
+        alt="icon-search"
+        src="@/assets/icons/search.svg"
+      />
+    </div>
+    <div
+      :class="searchQuery ? 'visible' : 'hidden'"
+      @click="searchQuery = null"
+    >
+      <img
+        svg-inline
+        class="icon icon-close"
+        alt="icon-close"
+        src="@/assets/icons/close.svg"
+      />
+    </div>
+  </div>
+</template>
+<script>
+import { ref } from "vue";
+export default {
+  name: "SearchBar",
+  props: {
```

Comment:
Do we need them as props? If you create a component which can be parameterized differently throughout the application then props will makes sense, otherwise you could stick to data.

Reply:
But what if it will be in the future? Shouldn´t we build our components with reusability in mind?

Reply:
This is where the YAGNI principle comes into the picture. If you know for sure, right now that you will need it in the future, then sure but if you are not sure, then you shouldn't add any (so called 'accidental') complexity.